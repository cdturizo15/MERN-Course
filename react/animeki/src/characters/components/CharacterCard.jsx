import React from 'react'
import { NavLink } from 'react-router-dom'

export const CharacterCard = ({ character }) => {
    const name = character.name.replace(/\s/g, '')

    return (
        <div className='card-container'>
            <NavLink className='nav-link' to={'/character/'+character.id}>
                <div className="card m-3" >
                    <img className="card-img-top profile-img" alt={character.name} src={'/src/assets/' + name + '.jpeg'} />
                    <div className="card-body">
                        <h3 className='card-title text-center'>{character.name}</h3>
                        <h5 className='text-center text-secondary'>{character.animeName}</h5>
                    </div>
                </div>
            </NavLink>

        </div>
    )
}
