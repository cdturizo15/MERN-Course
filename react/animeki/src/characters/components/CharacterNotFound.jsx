import React from 'react'

export const CharacterNotFound = () => {
  return (
    <div className='container-fluid d-flex justify-content-center align-content-center'>
        <h1>Character not found :(</h1>
    </div>
  )
}
