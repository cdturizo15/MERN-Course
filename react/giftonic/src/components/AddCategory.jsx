import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({setCategories})=>{
    const [input, setInput] = useState('');
    const handleChange = () =>{
        event.preventDefault();
        setCategories(input.toLowerCase())    }
    return(
        <div className="container">
            <form aria-label="form" onSubmit={handleChange}>
                <input onChange={(e)=> setInput(e.target.value)} placeholder="Buscar gif" className="input" type="text" />            </form>   
        </div>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}