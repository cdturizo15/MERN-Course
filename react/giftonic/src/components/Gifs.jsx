import PropTypes from 'prop-types'
export const Gifs = ({title, url}) => (
    <div className='card' >
            <img className='img-card' src={url} />            
    </div>
)

Gifs.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}