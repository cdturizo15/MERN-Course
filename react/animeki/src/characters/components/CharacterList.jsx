import React, { useMemo } from 'react'
import { getCharactersByGender } from '../helpers/getCharactersByGender'
import { CharacterCard } from './CharacterCard'

export const CharacterList = ({ gender, name }) => {
  const characters = useMemo(() => getCharactersByGender(gender), [gender]) 
  return (

    <div className='cards-container'>
      {characters.map(character => {
        return (<CharacterCard character={character} key={character.id} />)
      })}
    </div>


  )
}
