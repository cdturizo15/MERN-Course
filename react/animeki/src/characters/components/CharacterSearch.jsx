import React, { useMemo } from 'react'
import { getCharactersByName } from '../helpers/getCharacterByName'
import { CharacterCard } from './CharacterCard'
import { CharacterNotFound } from './CharacterNotFound'

export const CharacterSearch = ({ name }) => {
  const characters = useMemo(() => getCharactersByName(name), [name]) 
  return (

    <div className='cards-container'>

      {characters.length == 0 ? <CharacterNotFound></CharacterNotFound> : characters.map(character => {
        return (<CharacterCard character={character} key={character.id} />)
      })}
    </div>


  )
}
