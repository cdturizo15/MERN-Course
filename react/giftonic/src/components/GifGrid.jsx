
import { Gifs } from './Gifs';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import PropTypes from 'prop-types'


export const GifGrid = ({category}) =>{

    const {images, isLoading} = useFetchGifs (category)
    
    return (
        <>
        {
            isLoading && <h1>Is loading...</h1>
        }     
        {

        images.map((img) => (
            <Gifs {...img} key={img.id}/>
        ))
        
        }
         
        </>
        
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}