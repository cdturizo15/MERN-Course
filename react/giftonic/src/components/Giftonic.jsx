import { useState } from "react"
import { AddCategory } from './AddCategory';
import { GifGrid } from "./GifGrid";

export const Giftonic = () =>{
    const [categories, setCategories] = useState(['gifs'])
    const newCategory = (value)=>{
        if(value.trim().length <= 1) return;
        if( categories.includes(value) )return;
        setCategories([value,...categories])
        
    }
    const handleErase = (key)=>{
        const newCategories = categories.filter(cat => cat!= key)
        setCategories(newCategories)
    }
    const handleSearch = (key)=>{
        const newCategories = categories.filter(cat => cat!= key)
        setCategories(key, ...newCategories)
    }
    return(
        <div className="main-container">
            <h1>Giftonic</h1>
            <AddCategory setCategories= {newCategory} />
            <p className="recent-search-title">Recent searchs:</p>
            <div className="container-categories">
            {categories.map(cat=>(
                <p className="category" key={cat}><span onClick={()=>handleSearch(cat)} className="category-span">{cat}</span><i onClick={()=>handleErase(cat)} className="remove-icon">x</i></p>
                                
            ))}
            </div>
            <div className="gifs-container">
            {categories.map((category) =>{

                return (
                    <GifGrid category={category} key={category}/>
                )
            })}
            </div>
                      
        </div>
    )
}
