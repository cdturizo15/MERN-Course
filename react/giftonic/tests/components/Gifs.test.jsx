import { render } from "@testing-library/react"
import { Gifs } from "../../src/components/Gifs"
import { screen } from "@testing-library/react"

describe('Pruebas en <Gifs />', () => {

    const title = "saitama"
    const url = "https://one-punch.com/saitama.jpg"
    test('debe hacer match con snapshot', () => { 
        const {container} = render(<Gifs title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar imagen con url ', () => {
        render(<Gifs title={title} url={url} />)
        //screen.debug();
        //expect( screen.getByRole('img').src ).toBe(url);
        const {src, alt} = screen.getByRole('img')
        expect( src ).toBe(url)
        
    })
})