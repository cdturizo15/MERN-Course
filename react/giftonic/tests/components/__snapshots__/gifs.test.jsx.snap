// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Gifs /> debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      class="img-card"
      src="https://one-punch.com/saitama.jpg"
    />
  </div>
</div>
`;
